intf/locus.ml

```diff
@@ -15,6 +15,7 @@ open Misctypes
 
 type 'a occurrences_gen =
   | AllOccurrences
+  | AtLeastOneOccurrence
   | AllOccurrencesBut of 'a list (** non-empty *)
```

Comment:
I'm surprised this doesn't affect more tactics relying on this type. Does it make always make sense?

Reply:
Most of the tactics don't care that AllOccurrences doesn't catch any occurrence, and rely on ``progress`` instead to check that. It's useful for higher-order matching though, e.g. for "rewrite" we want the lhs to have at least one occurrence in the goal.